earthfile2llb/listener.go

```diff
@@ -662,6 +667,12 @@ func (ssf *StringSliceFlag) Set(arg string) error {
 	return nil
 }
 
+var shellNameStringRegexp = regexp.MustCompile("^[a-zA-Z_]+[a-zA-Z0-9_]*")
+
+func checkShellName(str string) string {
+	return shellNameStringRegexp.FindString(str)
```

Comment:
Can you make this func return a `bool`? Or even better - an `error`? You'll probably need to use `regexp.MatchString`.